controller/node_controller.go

```diff
@@ -995,6 +1001,169 @@ func (nc *NodeController) syncNodeStatus(pod *corev1.Pod, node *longhorn.Node) e
 	return nil
 }
 
+func (nc *NodeController) environmentCheck(kubeNode *corev1.Node, node *longhorn.Node) {
+	nc.syncKernelVersion(kubeNode, node)
+	namespaces := []lhtypes.Namespace{lhtypes.NamespaceMnt, lhtypes.NamespaceNet}
+	nc.syncPackagesInstalled(kubeNode, node, namespaces)
+	nc.syncMultipathd(node, namespaces)
+	nc.syncNFSClientVersion(kubeNode, node, namespaces)
+}
+
+func (nc *NodeController) syncKernelVersion(kubeNode *corev1.Node, node *longhorn.Node) {
+	kernelVersion := kubeNode.Status.NodeInfo.KernelVersion
+	brokenKernelVersion := []string{"5.15.0", "6.5.6"}
+	majorKernel5150Version := 5
+	majorKernel656Version := 6
+	brokenKernel5150PatchVersion := 94
+	brokenKernel656PatchVersion := 6
+	fixedKernel5150PatchVersion := 100
+
+	for _, version := range brokenKernelVersion {
+		if !strings.Contains(kernelVersion, version) {
+			continue
+		}
+		var kernel, major, minor, patch int
+		if _, err := fmt.Sscanf(kernelVersion, "%d.%d.%d-%d", &kernel, &major, &minor, &patch); err != nil {
```

Comment:
Some distros don't have `-%d`. Suggest using regexp.
An example https://goplay.tools/snippet/EyotP7UfLSy

Reply:
Used `regex` to parse the kernel version. Thanks.